lib/memory_profiler/reporter.rb

```diff
@@ -100,12 +106,19 @@ def object_list(generation)
           class_name = helper.lookup_class_name(klass)
           gem        = helper.guess_gem(file)
 
-          string     = '' << obj  if klass == String
+          if klass == String
+            if mem_lite
+              string = helper.string_summary(ObjectSpace._id2ref(obj.__id__))
```

Comment:
You could just pass `obj`. You don't need `ObjectSpace._id2ref(obj.__id__)`.

Reply:
Cool.  Wasn't positive how that all worked under the hood, and probably was under some impression that the `obj` object was some kind of clone of the object (which really doesn't make sense in hindsight).

Will change.